floor_generator/lib/writer/query_method_writer.dart

```diff
@@ -126,7 +126,10 @@ class QueryMethodWriter implements Writer {
     @nullable final String arguments,
     @nonNull final String mapper,
   ) {
-    final parameters = StringBuffer()..write("'${_queryMethod.query}', ");
+    final parameters = StringBuffer()
+      ..write(literalString(_queryMethod.query))
```

Comment:
> did you check out Dart's raw strings marked by a leading r? e.g. `r'string'`

No, but I definitely should, thanks! Edit: I did. 

> what if we would get rid of the multi to single line parsing which would allow people to use double quotes and thereby, would solve the initial problem described in #352?

What do you mean?

Reply:
We currently transform any query supplied to `@Query()` into a single-line string using single quotes. Even when the initial query was defined with double quotes, we get rid of them.

Now to the problem mentioned in #352. The user defined a multiline query with three leading and trailing single quotes. Using single quotes in such a string to define an empty string isn't a problem at all. The same applies to strings surrounded by double quotes.

As you found a solution for the problem, I suggest to keep the multi to single-line transformation issues in mind and get this PR merged without these changes.